plugin-server/tests/main/process-event.test.ts

```diff
@@ -1208,6 +1211,120 @@ test('snapshot event stored as session_recording_event', () => {
         window_id: undefined,
     })
 })
+const sessionReplayEventTestCases: {
+    snapshotData: { events_summary: RRWebEventSummary[] }
+    expected: Pick<
+        SummarizedSessionRecordingEvent,
+        'click_count' | 'keypress_count' | 'mouse_activity_count' | 'first_url' | 'first_timestamp' | 'last_timestamp'
+    >
+}[] = [
+    {
+        snapshotData: { events_summary: [{ timestamp: 1682449093469, type: 3, data: { source: 2 } }] },
+        expected: {
+            click_count: 1,
+            keypress_count: 0,
+            mouse_activity_count: 1,
+            first_url: undefined,
+            first_timestamp: '2023-04-25 18:58:13.469',
+            last_timestamp: '2023-04-25 18:58:13.469',
+        },
+    },
+    {
+        snapshotData: { events_summary: [{ timestamp: 1682449093469, type: 3, data: { source: 5 } }] },
+        expected: {
+            click_count: 0,
+            keypress_count: 1,
+            mouse_activity_count: 1,
+            first_url: undefined,
+            first_timestamp: '2023-04-25 18:58:13.469',
+            last_timestamp: '2023-04-25 18:58:13.469',
+        },
+    },
+    {
+        snapshotData: {
+            events_summary: [
+                {
+                    timestamp: 1682449093693,
+                    type: 5,
+                    data: {
+                        payload: {
+                            // doesn't match because href is nested in payload
+                            href: 'http://127.0.0.1:8000/home',
+                        },
+                    },
+                },
+                {
+                    timestamp: 1682449093469,
+                    type: 4,
+                    data: {
+                        href: 'http://127.0.0.1:8000/second/url',
+                    },
+                },
+            ],
+        },
+        expected: {
+            click_count: 0,
+            keypress_count: 0,
+            mouse_activity_count: 0,
+            first_url: 'http://127.0.0.1:8000/second/url',
+            first_timestamp: '2023-04-25 18:58:13.469',
+            last_timestamp: '2023-04-25 18:58:13.693',
+        },
+    },
+]
+sessionReplayEventTestCases.forEach(({ snapshotData, expected }) => {
```

Comment:
fwiw jest has `test.each`: https://dev.to/bgord/simplify-repetitive-jest-test-cases-with-test-each-310m

Maybe you know that but this was better, but just in case..